affected:
- package:
    ecosystem: PyPI
    name: tensorflow-cpu
    purl: pkg:pypi:tensorflow-cpu
  ranges:
  - events:
    - introduced: '0'
    - fixed: 8a6e874437670045e6c7dc6154c7412b4a2135e2
    repo: https://github.com/tensorflow/tensorflow
    type: GIT
  - events:
    - introduced: 2.3.0
    - fixed: 2.3.4
    - introduced: 2.4.0
    - fixed: 2.4.3
    type: ECOSYSTEM
aliases:
- CVE-2021-37644
- GHSA-27j5-4p9v-pp67
details: TensorFlow is an end-to-end open source platform for machine learning. In
  affected versions providing a negative element to `num_elements` list argument of
  `tf.raw_ops.TensorListReserve` causes the runtime to abort the process due to reallocating
  a `std::vector` to have a negative number of elements. The [implementation](https://github.com/tensorflow/tensorflow/blob/8d72537c6abf5a44103b57b9c2e22c14f5f49698/tensorflow/core/kernels/list_kernels.cc#L312)
  calls `std::vector.resize()` with the new size controlled by input given by the
  user, without checking that this input is valid. We have patched the issue in GitHub
  commit 8a6e874437670045e6c7dc6154c7412b4a2135e2. The fix will be included in TensorFlow
  2.6.0. We will also cherrypick this commit on TensorFlow 2.5.1, TensorFlow 2.4.3,
  and TensorFlow 2.3.4, as these are also affected and still in supported range.
id: PYSEC-0000-CVE-2021-37644
modified: '2021-08-27T03:22:43.455188Z'
published: '2021-08-12T21:15:00Z'
references:
- type: ADVISORY
  url: https://github.com/tensorflow/tensorflow/security/advisories/GHSA-27j5-4p9v-pp67
- type: FIX
  url: https://github.com/tensorflow/tensorflow/commit/8a6e874437670045e6c7dc6154c7412b4a2135e2
